### PR TITLE
Add basic secret manager frontend

### PR DESCRIPTION
**Pull request by alvaroaleman**
This PR adds a basic secret manager frontend, screenshots below. What is still missing is:
* Delete collections (needs an additional delete endpoint)
* Edit collection members (needs an "list all users" endpoint)

Ref https://issues.redhat.com/browse/DPTP-1891
/label tide/merge-method-squash

empty_initial
create_dialog
create_error
create_success

### CODE REVIEW

File: cmd/vault-secret-collection-manager/README.md [hunk 1]
```diff
@@ -25,3 +27,10 @@ Endpoints:
 * `GET /secretcollection`: Returns a list of all secret collections for the current user
 * `PUT /secretcollection/:name`: Creates a new secret collection using the provided `name`. The secret collection must not exist yet.
 * `PATCH /secretcollection/:name`: Changes the members of an existing secret colltion. The requesting user must be a member of the collection.
+
+## Development
+
+* Use `docker-compose` to start, vault, an oauth2 proxy and a dex instance as an IDP: `cd cmd/vault-secret-collection-manager && docker-compose up`
```
Comment (stevekuznetsov) [inline cmd/vault-secret-collection-manager/README.md:33]:
```
$ docker-compose up
bash: docker-compose: command not found...
```

:(

Any nontrivial way to get this working on normal CentOS/Fedora?

Comment (alvaroaleman) [reply cmd/vault-secret-collection-manager/README.md:33]: https://github.com/containers/podman-compose

**Comment by alvaroaleman** [comment]
Preview is at https://vault-secret-collection-manager-ci-tools-1747.apps.ci.l2s4.p1.openshiftapps.com

File: cmd/vault-secret-collection-manager/main.go [hunk 2]
```diff
@@ -135,6 +137,9 @@ func (c *idNameCache) set(name, id string) {
 
 func (m *secretCollectionManager) mux() *httprouter.Router {
 	router := httprouter.New()
+	router.GET("/", redirectHandler("/secretcollection?ui=true"))
```
Comment (stevekuznetsov) [inline cmd/vault-secret-collection-manager/main.go:140]: Why not a separate handler?

Comment (alvaroaleman) [reply cmd/vault-secret-collection-manager/main.go:140]: Mostly simplicity, it needs the same data, it just returns the UI instead of the raw data

**Pull request merged**